include/podio/CollectionBuffers.h

```diff
@@ -41,7 +42,9 @@ struct CollectionWriteBuffers {
 };
 
 struct CollectionReadBuffers {
+  bool needsSchemaEvolution{false};
   void* data{nullptr};
+  void* data_oldschema{nullptr};
```

Comment:
I think these really shouldn't go here. We can always simply construct new buffers from evolving old data. The management of whether we need schema evolution and which evolution function to call can also go somewhere else. What we should / could store here is the schema version of the currently stored buffers.